### PR TITLE
Switch to FlatEventPromoter

### PR DESCRIPTION
**Pull request by kwalcock**

### CODE REVIEW

**Comment by kwalcock** [comment]
I renamed the previous EventPromoter to HierarchicalEventPromoter and replaced its use with FlatEventPromoter.  The former one can be removed if necessary.

File: core/src/main/scala/ai/lum/odinson/state/EventPromoter.scala [hunk 1]
```diff
@@ -40,13 +41,62 @@ class EventPromoter {
...
+        // Turn this back on for paranoia mode.
+        // {
+        //  val namesAndArgumentMetadatas = eventMatch.argumentMetadata.groupBy(_.name)
+        //  namesAndArgumentMetadatas.foreach { case (name, argumentMetadatas) =>
+        //    // Those with the same name should all be the same.
+        //    argumentMetadatas.distinct.length == 1
+        //  }
+        // }
+        val namesToPromote: Set[String] = eventMatch.argumentMetadata
+            .map(_.name)
```
Comment (BeckySharp) [inline core/src/main/scala/ai/lum/odinson/state/EventPromoter.scala:69]: correct me if I'm wrong, but don't we need to check the `.promote` attribute here?

**Comment by kwalcock** [comment]
Good eye.  It got lost in a shuffle.

**Comment by kwalcock** [comment]
And that wasn't all.

**Review by BeckySharp — APPROVED** [review]
Looks awesome, thanks for adding this one!

**Comment by BeckySharp** [comment]
I think I should merge this now, towards getting that "one true branch" for the state changes, any objections?

**Pull request merged**